File: 第3合集-AI时代编程新范式：Vibe Coding/260306-大模型编程能力实测/02-模型源码/Qwen-3.5/bilibili-platform/backend/src/routes/comment.js
const express = require('express');
const router = express.Router();
const { Bilibili } = require('../utils/bilibili');

router.get('/:bvid', async (req, res, next) => {
  try {
    const { bvid } = req.params;
    const api = new Bilibili(req.session.cookies);
    const comments = await api.getComments(bvid);
    res.json({ success: true, data: comments });
  } catch (error) {
    next(error);
  }
});

router.post('/send', async (req, res, next) => {
  try {
    const { bvid, message, replyId } = req.body;
    const api = new Bilibili(req.session.cookies);
    const result = await api.sendComment(bvid, message, replyId);
    res.json({ success: true, data: result });
  } catch (error) {
    res.status(401).json({ success: false, error: '需要登录' });
  }
});

router.post('/reply', async (req, res, next) => {
  try {
    const { bvid, rootId, message } = req.body;
    const api = new Bilibili(req.session.cookies);
    const result = await api.replyComment(bvid, rootId, message);
    res.json({ success: true, data: result });
  } catch (error) {
    res.status(401).json({ success: false, error: '需要登录' });
  }
});

router.post('/like', async (req, res, next) => {
  try {
    const { bvid, rpid, action } = req.body;
    const api = new Bilibili(req.session.cookies);
    const result = await api.likeComment(bvid, rpid, action);
    res.json({ success: true, data: result });
  } catch (error) {
    res.status(401).json({ success: false, error: '需要登录' });
  }
});

module.exports = router;

File: 第3合集-AI时代编程新范式：Vibe Coding/260306-大模型编程能力实测/02-模型源码/DeepSeek-V3.2/app/routes/comment.py
from fastapi import APIRouter
from fastapi.responses import JSONResponse

from bilibili_api import comment, video
from bilibili_api.comment import CommentResourceType, OrderType

router = APIRouter()


@router.get("/list/{bvid}")
async def get_comments(bvid: str, page: int = 1, size: int = 20):
    try:
        v = video.Video(bvid=bvid)
        oid = await v.get_aid()

        comments_data = await comment.get_comments(
            oid=oid,
            type_=CommentResourceType.VIDEO,
            page_index=page,
            page_size=size,
            order=OrderType.TIME,
        )

        comment_list = []
        if "replies" in comments_data:
            for c in comments_data["replies"]:
                comment_list.append(
                    {
                        "rpid": c.get("rpid"),
                        "mid": c.get("member", {}).get("mid"),
                        "uname": c.get("member", {}).get("uname"),
                        "face": c.get("member", {}).get("face"),
                        "message": c.get("content", {}).get("message", ""),
                        "like": c.get("like", 0),
                        "rcount": c.get("rcount", 0),
                        "ctime": c.get("ctime", 0),
                    }
                )

        return JSONResponse(
            {
                "success": True,
                "data": {
                    "total": comments_data.get("page", {}).get("count", 0),
                    "page": page,
                    "size": size,
                    "comments": comment_list,
                },
            }
        )
    except Exception as e:
        return JSONResponse({"success": False, "error": str(e)})


@router.post("/send/{bvid}")
async def send_comment(bvid: str, text: str, credential_data: dict = None):
    try:
        if not credential_data:
            return JSONResponse({"success": False, "error": "需要登录凭据"})

        from bilibili_api import Credential

        credential = Credential(
            sessdata=credential_data.get("sessdata"),
            bili_jct=credential_data.get("bili_jct"),
            buvid3=credential_data.get("buvid3"),
        )

        v = video.Video(bvid=bvid)
        oid = await v.get_aid()

        result = await comment.send_comment(
            text=text, oid=oid, type_=CommentResourceType.VIDEO, credential=credential
        )

        return JSONResponse(
            {
                "success": True,
                "data": {"rpid": result.get("rpid"), "message": "评论发送成功"},
            }
        )
    except Exception as e:
        return JSONResponse({"success": False, "error": str(e)})


@router.post("/reply/{bvid}")
async def reply_comment(
    bvid: str, text: str, root_rpid: int, parent_rpid: int, credential_data: dict = None
):
    try:
        if not credential_data:
            return JSONResponse({"success": False, "error": "需要登录凭据"})

        from bilibili_api import Credential

        credential = Credential(
            sessdata=credential_data.get("sessdata"),
            bili_jct=credential_data.get("bili_jct"),
            buvid3=credential_data.get("buvid3"),
        )

        v = video.Video(bvid=bvid)
        oid = await v.get_aid()

        result = await comment.send_comment(
            text=text,
            oid=oid,
            type_=CommentResourceType.VIDEO,
            root=root_rpid,
            parent=parent_rpid,
            credential=credential,
        )

        return JSONResponse(
            {
                "success": True,
                "data": {"rpid": result.get("rpid"), "message": "回复发送成功"},
            }
        )
    except Exception as e:
        return JSONResponse({"success": False, "error": str(e)})
